### PR TITLE
Add traction form to INSAD implementation

### LINKED ISSUE #15221: Add Traction form on INS AD kernels
## Reason
It would allow to solve for cases where the viscosity simplification cannot be assumed.

## Design
It would be nice to have the equivalent of the MomentumTractionForm and MomentumTractionFormRZ kernels in the AD versions, at least in the INSADMomentumViscous.

## Impact
It would really help solve for cases where the Laplace form cannot be assumed. In my particular case, I'm looking at solving cases where the viscosity is not constant, and so far I can do that with the MomentumTraction kernels but not the MomentumLaplace kernels. Not sure if it is even possible to make such an addition (the traction form) on the AD equivalent. Sadly not an expert.

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #15221 

Includes equivalence testing between hand-coded and AD implementations for both unstabilized and stabilized cases.

@jiangwen84 you should probably be using this new form for your work if you have a non-constant viscosity. The laplace form is strictly only valid when the viscosity is constant

### CODE REVIEW

**Comment by jiangwen84** [comment]
@lindsayad thanks. I will switch it to use this form.

File: modules/navier_stokes/src/bcs/INSADMomentumNoBCBC.C [hunk 1]
```diff
@@ -33,16 +38,25 @@ INSADMomentumNoBCBC::INSADMomentumNoBCBC(const InputParameters & parameters)
...
 {
   // The viscous term
-  // -mu * (grad(u).n) * test
-  ADReal residual = -_mu[_qp] * (_grad_u[_qp] * _normals[_qp]) * _test[_i][_qp];
+  ADReal residual;
+  if (_form == "laplace")
+    residual = -_mu[_qp] * (_grad_u[_qp] * _normals[_qp]) * _test[_i][_qp];
+  else
+    residual =
+        -_mu[_qp] * ((_grad_u[_qp] + _grad_u[_qp].transpose()) * _normals[_qp]) * _test[_i][_qp];
```
Comment (fdkong) [inline modules/navier_stokes/src/bcs/INSADMomentumNoBCBC.C:59]: `((_grad_u[_qp] + _grad_u[_qp].transpose()) * _normals[_qp])` needes to be multiplied by `0.5`? When considering `grad_u[_qp] ` "equals" `0.5*(_grad_u[_qp] + _grad_u[_qp].transpose())`

I did not do the math on this, and so I do not know for sure.

Comment (fdkong) [reply modules/navier_stokes/src/bcs/INSADMomentumNoBCBC.C:59]: You may already cancel out somewhere else.

File: modules/navier_stokes/test/tests/ins/RZ_cone/tests [hunk 2]
```diff
@@ -167,4 +175,27 @@
...
+    type = 'Exodiff'
+    input = 'rz_cone_by_parts_traction_steady_stabilized.i'
+    exodiff = 'rz_cone_by_parts_traction_steady_stabilized_out.e'
+    issues = '#14901'
+    requirement = 'The system shall be able to solve the steady incompressible Navier-Stokes equationswith SUPG and PSPG stabilization and a first order velocity basis using a hand-coded Jacobian, standard variable implementation in an RZ coordinate system while integrating the pressure term by parts, using a traction form for the viscous term, and applying a natural outflow boundary condition and reproduce the results of the AD, vector variable implementation.'
+    custom_cmp = 'rz-cone-with-globals.cmp' # To ignore the vector variable components automatically generated by libMesh/ParaView
+  []
+  [ad_rz_cone_by_parts_traction_steady_supg_pspg_jac]
+    type = 'PetscJacobianTester'
+    difference_tol = 1e-6
```
Comment (fdkong) [inline modules/navier_stokes/test/tests/ins/RZ_cone/tests:195]: It is magical

**Review by fdkong — APPROVED** [review]

File: modules/navier_stokes/src/bcs/INSADMomentumNoBCBC.C (hunk 1 above)
Comment (lindsayad) [reply modules/navier_stokes/src/bcs/INSADMomentumNoBCBC.C:59]: Look at equation 8 of our INS paper :smile:

File: modules/navier_stokes/test/tests/ins/RZ_cone/tests (hunk 2 above)
Comment (lindsayad) [reply modules/navier_stokes/test/tests/ins/RZ_cone/tests:195]: So nice not having to worry about Jacobians when doing complex vector operations and stabilization methods! Development is so fast!

File: modules/navier_stokes/src/bcs/INSADMomentumNoBCBC.C (hunk 1 above)
Comment (fdkong) [reply modules/navier_stokes/src/bcs/INSADMomentumNoBCBC.C:59]: Oh, I see! Thanks

**Comment by anicolas91** [comment]
This is amazing and elegant. AD is awesome. Thank you so much!

**Pull request merged**